components/tasks/cu_pid/src/lib.rs

```diff
@@ -229,7 +229,11 @@ where
     ) -> CuResult<()> {
         match input.payload() {
             Some(payload) => {
-                let tov = input.metadata.tov.unwrap();
+                let tov = match input.metadata.tov {
+                    Tov::Time(single) => single,
+                    _ => panic!("Unexpected variant"),
```

Comment:
Is this panicking because nothing should ever continue if this happens?

An early-return with Err would be more configurable for error handling

Reply:
indeed, fixed thanks